core/node/da_clients/src/avail.rs

```diff
@@ -0,0 +1,189 @@
+use std::{
+    fmt::{Debug, Formatter},
+    sync::Arc,
+};
+
+use alloy::{
+    primitives::{B256, U256},
+    sol,
+};
```

Comment:
`alloy` is only used for rather simple purposes here. We are hesitant to introduce it until we properly migrate.
May I ask you to reimplement this functionality with `zksync_basic_types::web3::contract::{Tokenizable, Tokenize}` (e.g. manually do what the proc macro does for you now?

I understand that it isn't strictly convenient, but if we will enter a phase where half of the code uses old `web3` code and half is on `alloy`, it'll be extremely messy.

Reply:
makes sense, it was part of the implementation done by Avail team, but it's related to Stage 2 (inclusion verification), which is out of the scope of this PR, so I removed it completely